packages/messaging/index.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * @module Messaging
```

Comment:
I know this is a pain but I think we should merge in the history here and additionally separate it from all the other changes here.

Please do push back if there's more than an hours work in splitting though.